src/plugins/intel_npu/src/plugin/include/metadata.hpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <stdint.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace intel_npu {
+
+constexpr std::string_view MAGIC_BYTES = "OVNPU";
+
+constexpr uint16_t make_version(uint8_t major, uint8_t minor) {
+    return ((0 | major) << sizeof(major) * 8) | minor;
```

Comment:
Not sure that I understand this one. I was expecting to see something as:
(major << 16 ) | ( minor & 0x0000ffff)
For 32 bits.

Reply:
This suggestion is much better. I made the changes.